src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -142,7 +142,7 @@ public override BoundNode VisitITuplePattern(BoundITuplePattern node)
         protected override LocalState VisitSwitchStatementDispatch(BoundSwitchStatement node)
         {
             // first, learn from any null tests in the patterns
-            int slot = MakeSlot(node.Expression);
+            int slot = node.Expression.IsSuppressed ? GetOrCreatePlaceholderSlot(node.Expression) : MakeSlot(node.Expression);
```

Comment:
Is the idea here that if `node.Expression` is suppressed it will get a new slot, but all suppressed expressions of the same form (i.e. minus suppression) will share the same slot, so there will be at most two slots for a given slot, a suppressed one and an unsuppressed one?

Reply:
I think what I'm asking is if this is the best place to fix this, as opposed to checking IsSuppressed in the helper for MakeSlot/GetOrCreateSlot.

Reply:
The intent of this fix is to reflect the fact that the input expression is copied for the purposes of the switch.  Since the `!` operator changes the relationship between the inferred state and the state of the original expression, we need to treat that copied expression as a new slot when suppressed.  I think this is the best place to do that since it is specific to switch.

---
In reply to: [358474870](https://github.com/dotnet/roslyn/pull/39697#discussion_r358474870) [](ancestors = 358474870)